src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs

```diff
@@ -381,6 +383,131 @@ public Task GetMigratedServerUrls()
+                var migrationConfiguration = JsonDeserializationServer.MigrationConfiguration(blittable);
+
+                if (string.IsNullOrWhiteSpace(migrationConfiguration.DatabaseTypeName))
+                    throw new ArgumentException("DatabaseTypeName cannot be null or empty");
+
+                if (string.IsNullOrWhiteSpace(migrationConfiguration.FullPathToMigrator))
+                    throw new ArgumentException("FullPathToMigrator cannot be null or empty");
+
+                if (migrationConfiguration.Input == null)
+                    throw new ArgumentException("Input cannot be null");
+
+                if (Directory.Exists(migrationConfiguration.FullPathToMigrator) == false)
+                    throw new InvalidOperationException($"Directory {migrationConfiguration.FullPathToMigrator} doesn't exist");
+
+                const string migratorFileName = "Raven.Migrator.dll";
+                var path = Path.Combine(migrationConfiguration.FullPathToMigrator, migratorFileName);
+                if (File.Exists(path) == false)
+                    throw new InvalidOperationException($"The file '{migratorFileName}' doesn't exist in path: {migrationConfiguration.FullPathToMigrator}");
+
+                var processStartInfo = new ProcessStartInfo
+                {
+                    FileName = "dotnet",
+                    Arguments = $"{migratorFileName} {migrationConfiguration.DatabaseTypeName}",
+                    WorkingDirectory = migrationConfiguration.FullPathToMigrator,
+                    CreateNoWindow = true,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    RedirectStandardInput = true,
+                    UseShellExecute = false
+                };
+
+                Process process = null;
+                try
+                {
+                    process = Process.Start(processStartInfo);
+                }
+                catch (Exception e)
+                {
+                    process?.Kill();
+                    throw new InvalidOperationException("Unable to execute Migrator." + Environment.NewLine +
+                                                        "Command was: " + Environment.NewLine +
+                                                        (processStartInfo.WorkingDirectory ?? Directory.GetCurrentDirectory()) + "> "
+                                                        + processStartInfo.FileName + " " + processStartInfo.Arguments, e);
+                }
+
+                await process.StandardInput.WriteLineAsync(migrationConfiguration.Input.ToString());
+
+                if (migrationConfiguration.IsExportCommand == false)
+                {
+                    try
+                    {
+                        var line = process.StandardOutput.ReadLine();
+                        // return type is json, will throw if we fail to deserialize
+                        JsonConvert.DeserializeObject(line);
```

Comment:
This seems a bit harsh to deserialize the JSON just for validation, I'm sure there are cheaper ways to achieve this.

Reply:
I don't like it either, this is just throwing a lot of perf out the window

Reply:
Better to pass it to the client and have the error there